File: backend/Testing/Browser/Page/ResetPasswordPage.cs
using Microsoft.Playwright;

namespace Testing.Browser.Page;

public class ResetPasswordPage : AuthenticatedBasePage<ResetPasswordPage>
{
    public ResetPasswordPage(IPage page)
    : base(page, "/resetPassword", page.GetByRole(AriaRole.Button, new() { Name = "Reset Password", Exact = true }))
    {
    }

    public async Task FillForm(string newPassword)
    {
        await Page.GetByLabel("New Password").FillAsync(newPassword);
    }

    public async Task<UserDashboardPage> Submit()
    {
        await Page.GetByRole(AriaRole.Button, new() { Name = "Reset Password", Exact = true }).ClickAsync();
        return await new UserDashboardPage(Page).WaitFor();
    }
}

File: frontend/tests/pages/resetPasswordPage.ts
import type { Page } from '@playwright/test';
import { AuthenticatedBasePage } from './authenticatedBasePage';
import { UserDashboardPage } from './userDashboardPage';

export class ResetPasswordPage extends AuthenticatedBasePage {
  constructor(page: Page) {
    super(page, page.getByRole('heading', {name: 'Reset Password', exact: true}), `/resetPassword`);
  }

  fillForm(newPassword: string): Promise<void> {
    return this.page.getByLabel('New Password').fill(newPassword);
  }

  async submit(): Promise<UserDashboardPage> {
    await this.page.getByRole('button', {name: 'Reset Password', exact: true}).click();
    return await new UserDashboardPage(this.page).waitFor();
  }
}
